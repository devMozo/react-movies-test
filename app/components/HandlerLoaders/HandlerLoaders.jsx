import React from 'react';
import './HandlerLoaders.scss';
import PropTypes from 'prop-types';

const HandlerLoaders = ({ loading, error, onTryAgain, children }) => {
    return (
        <section className="HandlerLoaders">
            {loading && <div className="HandlerLoaders__loader"></div>}
            {!loading && error && (
                <React.Fragment>
                    <p className="HandlerLoaders__error-msg"> We had some connection issues with the API :( </p>
                    <button className="HandlerLoaders__button" onClick={onTryAgain}>
                        {' '}
                        Try Again{' '}
                    </button>
                </React.Fragment>
            )}
            {!loading && !error && children}
        </section>
    );
};

HandlerLoaders.propTypes = {
    loading: PropTypes.bool,
    error: PropTypes.bool,
    onTryAgain: PropTypes.func,
    children: PropTypes.any,
};

export default React.memo(HandlerLoaders);
