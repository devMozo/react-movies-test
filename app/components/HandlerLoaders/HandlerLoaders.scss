.HandlerLoaders {
    text-align: center;

    .HandlerLoaders__loader {
        border: 5px solid #f3f3f3;
        border-top: 5px solid #333;
        border-radius: 50%;
        margin: 40px auto;
        width: 40px;
        height: 40px;
        animation: spin 2s linear infinite;
    }

    .HandlerLoaders__button {
        padding: 20px 40px;
        background-color: #333;
        color: white;
        border: 0;
        font-size: 16px;
        transition: 0.3s;
        cursor: pointer;

        &:hover {
            opacity: 0.6;
        }
    }

    .HandlerLoaders__error-msg {
        font-size: 16px;
        margin: 40px 0 20px;
        color: #666;
    }

    @keyframes spin {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }
}
