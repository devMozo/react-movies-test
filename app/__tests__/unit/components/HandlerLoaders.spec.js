import React from 'react';
import { shallow } from 'enzyme';
import HandlerLoaders from '../../../components/HandlerLoaders/HandlerLoaders';

describe('<HandlerLoaders />', () => {
    const mockedFn = jest.fn();
    const PROPS = { loading: false, error: false, onTryAgain: mockedFn, children: <p> Mocked </p> };
    let component;

    beforeEach(() => {
        component = shallow(<HandlerLoaders {...PROPS} />);
    });

    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

    it("should trigger the function when the user clicks on the 'Try Again' button when an error appears", () => {
        component = shallow(<HandlerLoaders {...PROPS} error={true} />);
        const button = component.find('.HandlerLoaders__button');
        button.simulate('click');

        expect(mockedFn.mock.calls.length).toBe(1);
    });

    it("shouldn't display the button if it's all ok", () => {
        const button = component.find('.HandlerLoaders__button');

        expect(button.exists()).toBeFalsy();
    });
});
