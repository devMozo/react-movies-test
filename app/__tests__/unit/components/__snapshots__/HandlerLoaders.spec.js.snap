// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HandlerLoaders /> should render correctly in "debug" mode 1`] = `
<section
  className="HandlerLoaders"
>
  <p>
     Mocked 
  </p>
</section>
`;
